src/test_utils/katana/genesis.rs

```diff
@@ -0,0 +1,319 @@
+use std::marker::PhantomData;
+use std::path::PathBuf;
+use std::{collections::HashMap, path::Path};
+
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet::contract_class::ContractClass;
+use ethers::signers::LocalWallet;
+use ethers::signers::Signer;
+use ethers::types::U256;
+use eyre::{eyre, Result};
+use katana_primitives::block::GasPrices;
+use katana_primitives::contract::{StorageKey, StorageValue};
+use katana_primitives::genesis::constant::DEFAULT_FEE_TOKEN_ADDRESS;
+use katana_primitives::genesis::json::{FeeTokenConfigJson, GenesisJson};
+use katana_primitives::{
+    contract::ContractAddress,
+    genesis::json::{GenesisClassJson, GenesisContractJson, PathOrFullArtifact},
+};
+use lazy_static::lazy_static;
+use rayon::prelude::*;
+use reth_primitives::{Address, B256};
+use starknet::core::types::contract::legacy::LegacyContractClass;
+use starknet::core::types::FieldElement;
+use starknet::core::utils::{get_contract_address, get_storage_var_address, get_udc_deployed_address, UdcUniqueness};
+use walkdir::WalkDir;
+
+lazy_static! {
+    static ref SALT: FieldElement = FieldElement::from_bytes_be(&[0u8; 32]).unwrap();
+}
+
+#[derive(Debug, Clone)]
+pub struct Uninitialized;
+#[derive(Debug, Clone)]
+pub struct Loaded;
+#[derive(Debug, Clone)]
+pub struct Initialized;
+
+#[derive(Debug, Clone)]
+pub struct KatanaGenesisBuilder<T> {
+    coinbase: FieldElement,
+    classes: Vec<GenesisClassJson>,
+    class_hashes: HashMap<String, FieldElement>,
+    contracts: HashMap<ContractAddress, GenesisContractJson>,
+    token_storage: HashMap<StorageKey, StorageValue>,
```

Comment:
what is token_storage for? for the native token balances?

Reply:
yes correct

Reply:
can we name it something more evocative? such as `native_token_balances`?